### PR TITLE
Gnupg => 2.4.0 plus update necessary dependencies

### PR DESCRIPTION
**Pull request by satmandu**
- Also fixes up packages to allow gnupg to build on i686 w/o sommelier.
- Clean ups...

Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=gnupg240 CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/libtiff.rb [hunk 1]
```diff
@@ -22,23 +22,24 @@ class Libtiff < Package
      x86_64: '33129073011fd853518e91376a84fa0ddbc61605c041297c0d130294a113f389'
   })
```
Comment (uberhacker) [inline packages/libtiff.rb:24]: Update to https://download.osgeo.org/libtiff/tiff-4.5.0.tar.xz?

File: packages/libheif.rb [hunk 2]
```diff
@@ -23,32 +23,29 @@ class Libheif < Package
      x86_64: '626d99ad78adb2e959bc0cf812ce53da110d79d3917378b110b914f545137660'
   })
```
Comment (uberhacker) [inline packages/libheif.rb:25]: Update to https://github.com/strukturag/libheif/releases/download/v1.15.1/libheif-1.15.1.tar.gz?

File: packages/libtiff.rb (hunk 1 above)
Comment (satmandu) [reply packages/libtiff.rb:24]: Building 4.5.0...

**Review by uberhacker — CHANGES_REQUESTED** [review]
Nice!  I like the patch removals.  The library updates should be quick, right?

**Comment by satmandu** [comment]
> Nice!  I like the patch removals.  The library updates should be quick, right?

Updates done...

**Review by uberhacker — APPROVED** [review]

**Pull request merged**